### ISSUE TITLE
High level rules inside Arrays don't return the index on the error message

### ISSUE THREAD

**Issue by mrbongiolo** [dry-rb/dry-validation#305]
Ex: 
`{:should_have_investment_rate_only=>["must be filled", "must be filled", "must be filled", "cannot be defined"], 2=>{:minimum_investment=>["is missing"]}}`

The rule `:should_have_investment_rate_only` return an array of error messages, but it's not possible to know if it was on item 1,2 or 42. It would be better if it worked like the other rules.

The schema was something like:
```
BATCH_SCHEMA = Dry::Validation.JSON do
  each do
    schema(OTHER_SCHEMA)
  end
end

OTHER_SCHEMA = Dry::Validation.JSON do
  required(:minimum_investment) #blah blah

  rule(:should_have_investment_rate_only) do
    # the rule
  end
end
```

**Comment by gabteles** [comment]
+1 for this. There's any update on this?

**Comment by solnic** [comment]
I'll address this in 0.11.0

**Comment by YauhenVasileusky** [comment]
Any progress on this?

**Comment by solnic** [comment]
No. I haven't got time to work on dry-validation because of rom 4.0 effort. I hope to get back to dry-v before the end of the year.

**Comment by flash-gordon** [comment]
This now depends on https://github.com/dry-rb/dry-schema/issues/22

**Comment by solnic** [comment]
@flash-gordon sorry, I changed milestone to 2.0.0 w/o realizing that you set 1.0.0. I don't want dry-rb/dry-schema#22 to block 1.0.0 release. Does this work for you? I mean, if we can get it done before 1.0.0 then great, obviously, but if we don't it should be fine and we should release 1.0.0 regardless. There's nothing wrong in releasing 2.0.0 soon after 1.0.0.

**Comment by flash-gordon** [comment]
@solnic I just bumped the milestone, didn't mean to assign/set it actually. As I said in some other issue, it's only my hope, not a requirement. Also, it doesn't have to be 2.0 if it doesn't break the public API and it looks like adding support for arrays does not do it.

**Comment by solnic** [comment]
@flash-gordon OK I cleared the milestone for now, it'll be done when it's done then ;)
